controllers/dependencies.go

```diff
@@ -0,0 +1,167 @@
+package controllers
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+
+	"os"
+
+	"path/filepath"
+
+	git "github.com/go-git/go-git/v5"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	yamlutil "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"sigs.k8s.io/controller-runtime/pkg/log"
```

Comment:
I suggest we take the preflight path and use logrus for logging. We can configure the logging options once and reuse it.

Reply:
About logging as I put on the PR message I would suggest we discuss on further iterations. Here I'm following what I found on other pieces of code on this project itself. I would suggest creating an issue to standardize logging through the whole project. And then we can certainly include logrus as the standard.